certs/certs.go

```diff
@@ -34,6 +35,11 @@ const (
 	duration365d = time.Hour * 24 * 365
 )
 
+// ClusterCertificatesSecretName returns the name of the certificates secret, given a cluster name
+func ClusterCertificatesSecretName(clusterName string) string {
+	return fmt.Sprintf("%s-certs", clusterName)
```

Comment:
This is not unique to this PR, and we have other places in master that have this issue, but I wanted to point it out: if the clusterName is already at the max length, then clusterName-certs will be too long and the apiserver will reject it.